### PR TITLE
feat: design changes in signature paged message section

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Make message section in signature collapsible and add copy option.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3477

## **Manual testing steps**

1. Go to test dapp
2. Submit signatures and check message section

## **Screenshots/Recordings**
Screenshot 2024-10-23 at 5 27 35 PM

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/confirm/info/row/row.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ export type ConfirmInfoRowProps = {
   copyEnabled?: boolean;
   copyText?: string;
   'data-testid'?: string;
+  collapsible?: boolean;
```
Comment (matthewwalsh0) [inline ui/components/app/confirm/info/row/row.tsx:43]: Can we not re-use the existing `ui/components/app/confirm/info/row/expandable-row.tsx`?

File: ui/components/app/confirm/info/row/row.tsx [hunk 2]
```diff
@@ -81,69 +82,96 @@ export const ConfirmInfoRow: React.FC<ConfirmInfoRowProps> = ({
...
+          />
+        )}
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.flexStart}
+          color={color}
+        >
+          <Box display={Display.Flex} alignItems={AlignItems.center}>
```
Comment (matthewwalsh0) [inline ui/components/app/confirm/info/row/row.tsx:142]: While it's nice to add configuration to the core row, this adds a lot of logic and complexity, so would it be better to create an alternate wrapper component that potentially inherits the same properties of row? Or a wrapper component?

Or at the very least, should we break this very large schema up with local components?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/components/app/confirm/info/row/row.tsx (hunk 1 above)
Comment (jpuri) [reply ui/components/app/confirm/info/row/row.tsx]: The designs are very different from expandable row.

File: ui/components/app/confirm/info/row/row.tsx (hunk 2 above)
Comment (jpuri) [reply ui/components/app/confirm/info/row/row.tsx:144]: Changes are required in info row requires to support collapsible row, it is hard to only wrap into row and make it collapsible.

Comment (matthewwalsh0) [reply ui/components/app/confirm/info/row/row.tsx:144]: In that case, could we split this for readability into some local components (in this file) such as `Container`, `CopyIcon`, `Tooltip`, `Content`, `CollapsibleIcon` etc?

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/images/icons/collapse.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="m422 64c5 0 9 2 13 5 7 8 7 19 0 26l-107 107 66 0c11 0 19 7 19 18 0 12-8 19-19 19l-110 0c-2 0-6 0-8-2-2 0-3-2-5-4-2-1-4-3-4-5-2-2-2-6-2-8l0-110c0-11 8-19 19-19 11 0 18 8 18 19l0 66 107-107c4-3 7-5 13-5z m-194 193l-110 0c-11 0-19 8-19 19 0 11 8 18 19 18l66 0-107 107c-7 7-7 19 0 26 8 7 19 7 26 0l107-107 0 66c0 11 7 19 18 19 11 0 19-8 19-19l0-110c0-2 0-6-2-8 0-1-2-3-4-5-2-2-3-4-5-4-2-2-6-2-8-2z"/></svg>
```
Comment (georgewrmarshall) [inline app/images/icons/collapse.svg:1]: Just pushed a commit to optimize the svg here. https://github.com/MetaMask/metamask-extension/pull/28038/commits/44ec97d0b8139c0655a61c559ac8134ef3ab4fa7

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀 
- Checked collapse icon svg and that it renders in storybook correctly ✅ 
- Checked collapse icon exists in [Figma icon library](https://www.figma.com/design/cr3Bcm1TT1OAn0VNJaSBC1/DS-Icons-%26-Visual-Assets?m=auto&node-id=2176-62) for 1:1 match ✅ 
- Did not check any other part of the code ❌

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**